File: iosApp/iosApp/UI/Components/AppHeaderIcon.swift
import SwiftUI

/// Header dengan ikon bulat + title + subtitle opsional. Dipakai di puncak layar
/// Login / Register / walkthrough completed.
struct AppHeaderIcon: View {
    let systemImage: String
    let title: String
    let subtitle: String?

    var iconSize: CGFloat = AppSizes.headerIcon
    var iconFontSize: CGFloat = 42

    init(
        systemImage: String,
        title: String,
        subtitle: String? = nil,
        iconSize: CGFloat = AppSizes.headerIcon,
        iconFontSize: CGFloat = 42
    ) {
        self.systemImage = systemImage
        self.title = title
        self.subtitle = subtitle
        self.iconSize = iconSize
        self.iconFontSize = iconFontSize
    }

    var body: some View {
        VStack(spacing: AppSpacing.md) {
            ZStack {
                Circle()
                    .fill(Color.accentColor.opacity(0.12))
                    .frame(width: iconSize, height: iconSize)

                Image(systemName: systemImage)
                    .font(.system(size: iconFontSize, weight: .semibold))
                    .foregroundColor(.accentColor)
            }
            .padding(.top, AppSpacing.md)

            VStack(spacing: AppSpacing.xs) {
                Text(title)
                    .font(.title2.weight(.bold))
                    .foregroundColor(.primary)

                if let subtitle {
                    Text(subtitle)
                        .font(.subheadline)
                        .foregroundColor(.secondary)
                        .multilineTextAlignment(.center)
                }
            }
        }
    }
}

File: androidApp/src/main/kotlin/org/kmptemplate/project/ui/components/AppHeaderIcon.kt
package org.kmptemplate.project.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.sp
import org.kmptemplate.project.ui.theme.AppSizes
import org.kmptemplate.project.ui.theme.AppSpacing

/**
 * Header ikon bulat + title + subtitle. Digunakan di puncak layar Login/Register/Success.
 *
 * Emoji dipakai sebagai plceholder ikon supaya sederhana; kalau nanti diganti
 * SF-symbol-alike / vector, cukup ganti body-nya, tidak menyentuh call site.
 */
@Composable
fun AppHeaderIcon(
    emoji: String,
    title: String,
    modifier: Modifier = Modifier,
    subtitle: String? = null,
    iconSize: Dp = AppSizes.headerIcon,
    emojiFontSize: TextUnit = 38.sp,
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Box(
            modifier = Modifier
                .size(iconSize)
                .clip(CircleShape)
                .background(MaterialTheme.colorScheme.primaryContainer),
            contentAlignment = Alignment.Center
        ) {
            Text(text = emoji, fontSize = emojiFontSize)
        }
        Spacer(modifier = Modifier.height(AppSpacing.xl - AppSpacing.xs))
        Text(
            text = title,
            style = MaterialTheme.typography.headlineMedium.copy(fontWeight = FontWeight.Bold),
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onBackground
        )
        if (subtitle != null) {
            Spacer(modifier = Modifier.height(AppSpacing.sm))
            Text(
                text = subtitle,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center
            )
        }
    }
}
